src/utils/local_folder_path.ts

```diff
@@ -0,0 +1,32 @@
+import { statSync } from 'fs';
+import { basename, dirname, resolve } from 'path';
+
+export type FilePathAndName = [string, string?]; // TODO: consolidate with the download-file functionality
+
+/**
+ * Resolves the path, verifies its existance and type to conditionally return its dir and file name
+ * @param {string} destOutputPath - the path from where to extract the dir path and name
+ * @returns {FilePathAndName} - the directory where to put the file and the file name (which is undefined when the provided destOutputPath is a directory)
+ */
+export function getOutputFolderPathAndName(destOutputPath: string): FilePathAndName {
```

Comment:
TODO: add unit tests for this method
Q: How do I stub `resolve`, `dirname`,`basename`, `statSync` and `Stats.isDirectory` here?

Reply:
Do I inject `fs` or an fs-like interface?

Reply:
The approach was to use injection as said: 6112bd8